code-challenges.test.js

```diff
@@ -10,52 +12,127 @@
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
-
-
+// describe a function called encrypter that takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
+describe("encrypter", () => {
+    let secretCodeWord1 = "Lackadaisical"
+    let expectedOutput1 = "L4ck4d41s1c4l"
+    let secretCodeWord2 = "Gobbledygook"
+    let expectedOutput2 = "G0bbl3dyg00k"
+    let secretCodeWord3 = "Eccentric"
+    let expectedOutput3 = "3cc3ntr1c"
+    it("takes in a string and returns a coded message", () => {
+        expect(encrypter(secretCodeWord1)).toEqual(expectedOutput1)
+        expect(encrypter(secretCodeWord2)).toEqual(expectedOutput2)
+        expect(encrypter(secretCodeWord3)).toEqual(expectedOutput3)
+    })
+})
+
+// RED
 
 // b) Create the function that makes the test pass.
 
+// Create a function that takes in a string and uses .replace to change the values
+const encrypter = (str) => {
+    return str.replace(/[a]/gi, '4').replace(/[e]/gi, '3').replace(/[i]/gi, '1').replace(/[o]/gi, '0')
+}
 
+// GREEN
 
 // --------------------2) Create a function that takes in an array of words and a single letter and returns all the words that contain that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
-const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
-const letterA = "a"
-// Expected output: ["Apple", "Banana", "Orange"]
-const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
-const letterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
-
-
+// describe a function called doesContain that  takes in an array of words and a single letter and returns all the words that contain that particular letter.
+describe("doesContain", () => {
+    let arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
+    let letterA = "a"
+    let expectedOutput1 = ["Apple", "Banana", "Orange"]
+    let arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+    let letterE = "e"
+    let expectedOutput2 = ["Cherry", "Blueberry", "Peach"]
+        it(" takes in an array of words and a single letter and returns all the words that contain that particular letter.", () => {
+        expect(doesContain(arrayOfWords1, letterA)).toEqual(expectedOutput1)
+        expect(doesContain(arrayOfWords2, letterE)).toEqual(expectedOutput2)
+    })
+})
+
+// RED
 
 // b) Create the function that makes the test pass.
+// define doesContain with an input of an array and a single letter
 
+const doesContain = (check, letter) => {
+    let theseContain = []
+    for (let j = 0; j < check.length; j++) {
+        // if the word contains the letter in any case format, the original unmodified word is pushed onto the array and then later returned
+        if (check[j].toLowerCase().includes(letter.toLowerCase())){
+            theseContain.push(check[j])
+        }
+    }
+    return theseContain
+}
```

Comment:
This seems to be a sufficient solution but you can reduce the amount of code you are writing using something like `.filter`